ios/browser/api/features/features.mm

```diff
@@ -0,0 +1,960 @@
+#include "brave/components/brave_sync/features.h"
+#include "brave/components/brave_wallet/common/buildflags.h"
+#include "brave/components/brave_wallet/common/features.h"
+#include "brave/components/de_amp/common/features.h"
+#include "brave/components/debounce/common/features.h"
+#include "brave/components/ipfs/buildflags/buildflags.h"
+#include "brave/components/ipfs/features.h"
+#include "brave/components/ntp_background_images/browser/features.h"
+#include "brave/components/p3a/features.h"
+#include "brave/components/playlist/common/features.h"
+#include "brave/components/skus/common/features.h"
+#import "build/blink_buildflags.h"
+#include "build/build_config.h"
+#include "ios/chrome/browser/crash_report/model/features.h"
+#include "ios/chrome/browser/credential_provider_promo/model/features.h"
+#include "ios/chrome/browser/iph_for_new_chrome_user/model/features.h"
+#include "ios/chrome/browser/metrics/features.h"
+#include "ios/chrome/browser/ntp/features.h"
+#include "ios/chrome/browser/ntp/home/features.h"
+#include "ios/chrome/browser/parcel_tracking/features.h"
+#include "ios/chrome/browser/promos_manager/features.h"
+#include "ios/chrome/browser/safe_browsing/model/features.h"
+#include "ios/chrome/browser/screen_time/model/features.h"
+#include "ios/chrome/browser/shared/public/features/features.h"
+#include "ios/chrome/browser/snapshots/model/features.h"
+#include "ios/chrome/browser/tabs/model/features.h"
+#include "ios/chrome/browser/tabs/model/inactive_tabs/features.h"
+#include "ios/chrome/browser/tabs/model/tab_pickup/features.h"
+#include "ios/chrome/browser/ui/app_store_rating/features.h"
+#include "ios/chrome/browser/ui/default_promo/post_restore/features.h"
+#include "ios/chrome/browser/ui/download/features.h"
+#include "ios/chrome/browser/ui/infobars/coordinators/features.h"
+#include "ios/chrome/browser/ui/settings/privacy/privacy_guide/features.h"
+#include "ios/chrome/browser/web/features.h"
+#include "ios/chrome/browser/web/model/features.h"
+#include "ios/web/common/features.h"
+
+@interface Feature () {
+  const base::Feature* _value;
+}
+@end
+
+@implementation Feature
+- (instancetype)initWithName:(NSString*)name value:(const base::Feature*)value {
+  if ((self = [super init])) {
+    _name = name;
+    _value = value;
+  }
+  return self;
+}
+
+- (bool)enabled {
+  return base::FeatureList::IsEnabled(std::cref(*_value));
+}
+
+//- (void)setEnabled:(bool)enabled {
+//  std::vector<base::FeatureList::FeatureOverrideInfo> overrides = {
+//    {
+//        std::cref(*_value), enabled ?
+//        base::FeatureList::OverrideState::OVERRIDE_ENABLE_FEATURE :
+//        base::FeatureList::OverrideState::OVERRIDE_DISABLE_FEATURE
+//    }
+//  };
+//  base::FeatureList::GetInstance()->RegisterExtraFeatureOverrides(overrides);
+//}
+@end
+
+@implementation FeatureList
++ (Feature*)kARKillSwitch {
+  return [[Feature alloc] initWithName:@"kARKillSwitch" value:&kARKillSwitch];
+}
+
++ (Feature*)kAddressInfobarDisplayLength {
```

Comment:
these declarations can also be wrapped with a macro to simplify things.

Reply:
I'd prefer not to do that because this is an interface exported to Swift and C/C++ macros don't do very well in Swift.